docs/content/TopNQuery.md

```diff
@@ -211,4 +211,7 @@ Users who can tolerate *approximate rank* topN over a dimension with greater tha
     "queryType": "topN",
     "threshold": 2
 }
-```
\ No newline at end of file
+```
+
+### DimExtractionFn
```

Comment:
> All of that said, I think we definitely need the ability to do lookups on the broker as well to support the filter case. For filters that are on a namespaced dim extraction, I think we want the broker to do a single lookup and rewrite it into a large OR clause instead of doing the lookup on each historical.

Currently filtering is done via the extraction filter. Are you talking about having a reverse lookup? Note that in the current implementation a reverse lookup could be a one to many operation.

To go along with that, it would make sense to me to have reversibility be a property of the extraction function and have filters allowable on the dimension spec.

Be aware that the suggested broker-optimizations would also cause a query to a broker and a query to a historical node to return potentially drastically different results if not done properly. To carry over current behavior, I would expect a query to the historical node to return essentially a subset of what the broker would have returned.

Such an optimization on broker extraction function handling feels like it is significant scope creep for functionality of query time lookups, even though it could be a very good optimization.